diff --git a/src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri b/src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri
index 847f69c358..2360f6259e 100644
--- a/src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri
+++ b/src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri
@@ -234,3 +234,4 @@ SOURCES += \
     SOURCES += wtf/TCSystemAlloc.cpp
 }
 
+QMAKE_CXXFLAGS += -std=gnu++98
diff --git a/src/3rdparty/webkit/Source/WebKit.pri b/src/3rdparty/webkit/Source/WebKit.pri
index 01f88a485d..384b1ca349 100644
--- a/src/3rdparty/webkit/Source/WebKit.pri
+++ b/src/3rdparty/webkit/Source/WebKit.pri
@@ -199,3 +199,4 @@ CONFIG(qt_minimal) {
 }
 
 contains(DEFINES, QT_NO_UITOOLS): CONFIG -= uitools
+QMAKE_CXXFLAGS += -std=gnu++98
diff --git a/src/plugins/accessible/qaccessiblebase.pri b/src/plugins/accessible/qaccessiblebase.pri
index 95c1fad13a..76693d5fc5 100644
--- a/src/plugins/accessible/qaccessiblebase.pri
+++ b/src/plugins/accessible/qaccessiblebase.pri
@@ -1,2 +1,3 @@
 target.path     += $$[QT_INSTALL_PLUGINS]/accessible
 INSTALLS        += target
+QMAKE_CXXFLAGS += -std=gnu++98
